File: src/main/java/net/hectus/neobb/turn/default_game/flower/TRedTulip.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.RedstoneClazz;
import org.bukkit.block.Block;

import java.util.List;

public class TRedTulip extends FlowerTurn {
    public TRedTulip(NeoPlayer player) { super(player); }
    public TRedTulip(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public void apply() {
        player.game.allowedClazzes().add(RedstoneClazz.class);
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.ExtraTurn());
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/flower/TRedTulip.kt
package net.hectus.neobb.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.RedstoneClazz
import org.bukkit.block.Block

class TRedTulip(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 4

    override fun apply() {
        player!!.game.allowed.add(RedstoneClazz::class)
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(ExtraTurn())
    }
}
